core/state/statedb.go

```diff
@@ -1242,14 +1246,24 @@ func (s *StateDB) SlotInAccessList(addr common.Address, slot common.Hash) (addre
 	return s.accessList.Contains(addr, slot)
 }
 
+func (s *StateDB) GetTxHash() common.Hash {
+	return s.thash
+}
+
 // GetPredicateStorageSlots returns the storage slots associated with a given address, and whether or not
 // that address was included in the optional access list of the transaction.
 // The storage slots are returned in the same order as they appeared in the transaction.
 // These are the same storage slots that are used to verify any transaction
 // predicates for transactions with access list addresses that match a precompile address.
-func (s *StateDB) GetPredicateStorageSlots(address common.Address) ([]byte, bool) {
-	storageSlots, exists := s.predicateStorageSlots[address]
-	return storageSlots, exists
+func (s *StateDB) GetPredicateStorageSlots(address common.Address, index int) ([]byte, bool) {
+	predicates, exists := s.predicateStorageSlots[address]
+	if !exists {
+		return nil, false
+	}
+	if index >= len(predicates) {
```

Comment:
should we check if index is negative (or simply require uint64 indexes here)? 

this is related to the comment about lossless conversion between int<>int64 below.